eml/screens/courses/CourseScreen.jsx

```diff
@@ -1,72 +1,85 @@
-import { useRoute, useNavigation } from '@react-navigation/native'
+import { useNavigation } from '@react-navigation/native'
 import { React, useEffect, useState } from 'react'
-import CourseListUI from '../../components/easyDynComponents/courseListUI'
-import { View, Pressable, Text } from 'react-native'
+import { View, Pressable, Text, Dimensions, Image, ScrollView, StyleSheet } from 'react-native'
 import { useFonts, VarelaRound_400Regular } from '@expo-google-fonts/dev'
 import { AppLoading } from 'expo-app-loading'
+//import { ErrorBoundary } from 'react-error-boundary'
 import * as StorageService from "../../services/StorageService";
+import CourseCard from '../../components/courses/courseCard/CourseCard'
+import SectionScreen from '../section/sectionScreen'
 
 export default function CourseScreen() {
 
 
-    const route = useRoute();
+    const [courses, setCourses] = useState([]);
 
-    const [course, setCourse] = useState({});
-
-    const [courseLoaded, setCourseLoaded] = useState(false);
-
-    const [downloadState, setDownloadState] = useState(null);
-
-    let courseId = null
-    if (route.params !== undefined) {
-        courseId = route.params.courseId;
-    }
+    const [courseLoaded, setCourseLoaded] = useState();
 
     const navigation = useNavigation()
-    let currentCourse = null;
 
     let [fontsLoaded] = useFonts({
         VarelaRound_400Regular
     })
 
-    async function loadCourse() {
-        const courseData = await StorageService.getCourseById(courseId);
-        setCourse(courseData);
-    }
-
+    
+    
     useEffect(() => {
```

Comment:
Pretty sure this is bad. Seems to me changing the state using setCourses inside this useEffect creates an infinite loop? Please correct me if not.

Reply:
This should indeed be the case. Haven't tested it yet

Reply:
This is true

Reply:
Indeed this is bad. The only reason it is done this way is because the functions within StorageService is setup so that it apparently loads the courses after rendering the page. So the page does not show the courses until re-rendered. The quick and very very dirty fix was to have the infinite loop with use effect. When it renders the page in the second iteration of the loop, it does so with all the courses loaded from the first. All iterations of the loop afterwards serves no function. This was the "it have to work now" solution.